src/client/customers.rs

```diff
@@ -432,6 +437,203 @@ pub struct VoidInitiatedLedgerEntry {
     pub void_amount: serde_json::Number,
 }
 
+/// The view mode for a cost breakdown.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize_enum_str, Serialize_enum_str)]
+pub enum CostViewMode {
+    /// Provide results as an incremental day-by-day view.
+    #[serde(rename = "periodic")]
+    Periodic,
+    /// Provide results as cumulative totals since the start of the billing period.
+    #[serde(rename = "cumulative")]
+    Cumulative,
+}
+
+#[derive(Debug, Default, Clone)]
+struct CustomerCostParamsFilter<'a> {
+    timeframe_start: Option<OffsetDateTime>,
+    timeframe_end: Option<OffsetDateTime>,
+    view_mode: Option<CostViewMode>,
+    group_by: Option<&'a str>,
+}
+
+impl<'a> CustomerCostParamsFilter<'a> {
+    /// Apply the filter to a request.
+    fn apply(&self, mut req: RequestBuilder) -> RequestBuilder {
```

Comment:
it would be more canonical for this to be implemented on `RequestBuilder`, so you can:
```
let req = req.apply(filter).apply(other_filter).etc();
```